From: "Rafael J. Wysocki" <rafael.j.wysocki@intel.com>
Date: Sat, 18 Sep 2021 14:56:17 +0200
Subject: ACPI: glue: Drop cleanup callback from struct acpi_bus_type
Patch-mainline: v5.16-rc1
Git-commit: c4d19838d8c42828c7e39e4432ea5b33f02dac27
References: jsc#PED-1408

Since PCI was the only user of the ->cleanup callback in struct
acpi_bus_type and it is not using struct acpi_bus_type any more,
drop that callback from there and update acpi_device_notify_remove()
accordingly.

Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Tested-by: Ferry Toth <fntoth@gmail.com>
Acked-by: Lee, Chun-Yi <jlee@suse.com>
---
 drivers/acpi/glue.c     |   12 +++---------
 include/acpi/acpi_bus.h |    1 -
 2 files changed, 3 insertions(+), 10 deletions(-)

--- a/drivers/acpi/glue.c
+++ b/drivers/acpi/glue.c
@@ -337,16 +337,10 @@ void acpi_device_notify_remove(struct de
 	if (!adev)
 		return;
 
-	if (dev_is_pci(dev)) {
+	if (dev_is_pci(dev))
 		pci_acpi_cleanup(dev, adev);
-	} else {
-		struct acpi_bus_type *type = acpi_get_bus_type(dev);
-
-		if (type && type->cleanup)
-			type->cleanup(dev);
-		else if (adev->handler && adev->handler->unbind)
-			adev->handler->unbind(dev);
-	}
+	else if (adev->handler && adev->handler->unbind)
+		adev->handler->unbind(dev);
 
 	acpi_unbind_one(dev);
 }
--- a/include/acpi/acpi_bus.h
+++ b/include/acpi/acpi_bus.h
@@ -570,7 +570,6 @@ struct acpi_bus_type {
 	bool (*match)(struct device *dev);
 	struct acpi_device * (*find_companion)(struct device *);
 	void (*setup)(struct device *);
-	void (*cleanup)(struct device *);
 };
 int register_acpi_bus_type(struct acpi_bus_type *);
 int unregister_acpi_bus_type(struct acpi_bus_type *);
